Implemented random events

barnahadnagy opened the pull request:
- Code cleanup and bug fixes in turn handling and the database
- Effect parsers much stronger and can access any information about the country

Commit:
Changed cull units to randomly cull the appropriate amount of units.

Commit:
Added dynamic events

Commit:
Fixed some issues

- There is a bug with EF currently under investigation

Commit:
Fixed minor issues

- Added EF explanation
- Removed redundant code

koppa96 reviewed (changes requested):
Nice changes, some tests would be appreciated.

Commit:
Added tests, fixed some issues

- Moved a lot of known values into the globals
- Country now starts with two buildings
- Fixed some bugs in unit desertion and unit culling

barnahadnagy commented:
Added tests along with some bugfixes

barnahadnagy committed:
Merge branch 'master' into randomevent
